pennylane/compiler/python_compiler/transforms/quantum/split_non_commuting.py

```diff
@@ -0,0 +1,439 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This file contains a limited prototype of the split_non_commuting pass.
+
+Known Limitations
+-----------------
+
+  * Only single-term observables with no coefficients are supported - there is no support for CompositeOp or SymbolicOp observables
+  * Only the Expval measurement process is supported
+  * There is no option to specify a grouping strategy (this will be more relevant once CompositeOp support is added)
+  * Hence, only the "wires" grouping strategy is implemented, not taking into account observable-commutation logic yet.
+  * There is no efficient handling of duplicate observables - a circuit that returns multiple measurements on the same observable will split into multiple executions (this will be more relevant once CompositeOp support is added)
```

Comment:
Would it be possible to add some tips and thoughts on the postprocessing part as well? I guess we need to restore to order of the measurements later.